exampleCourse/questions/element/truthTable/question.html

```diff
@@ -0,0 +1,33 @@
+<pl-question-panel>
+<p>
+What is the resulting values of the expressions
+<pl-code language="python">y == A or B</pl-code>
+<pl-code language="python">y == C and D</pl-code>
+<pl-code language="python">y == D and E</pl-code>
+<pl-code language="python">y == X or Y</pl-code>
+given the values in the following truth tables?
+</p>
+</pl-question-panel>
+
+<pl-truth-table answers-name="answer1" fixed-order="true" true-label="true" false-label="false" num-rows="3">
+    {{#params.variables}}
+    <pl-variable>{{name}}</pl-variable>
+    {{/params.variables}}
+    <pl-answer-column><pl-code language="python">A or B</pl-code></pl-answer-column>
+    <pl-answer-column expression="C and D">C and D</pl-answer-column>
+</pl-truth-table>
+<pl-truth-table answers-name="answer2" true-label="on" false-label="off">
+    <pl-variable>C</pl-variable>
+    <pl-variable>D</pl-variable>
+    <pl-variable>E</pl-variable>
+    <pl-answer-column expression="D and E">D and E</pl-answer-column>
+    <pl-row>1 0 1</pl-row>
+    <pl-row>1 1 1</pl-row>
+    <pl-row>0 0 0</pl-row>
+    <pl-row>0 1 0</pl-row>
+</pl-truth-table>
```

Comment:
Everything of mine was a Python hack, so I don't have any nice HTML/element structure to compare to.

Is the purpose of those lines simply to limit the rows that show up?  So, only 4 of the 8 rows of that truth table will show up?

If so, maybe it would be easier to have a list attribute such as `show-rows` that took a list of integers and showed the corresponding rows (i.e., the rows with the corresponding binary representations).  In this example, someone could write `show-rows="5,7,0,2"`.

I guess this would be limiting in that you could not have rows with non-binary entries.  I don't recall doing that, but maybe somebody will want to use an `X` or a variable at some point. It doesn't seem likely to me, though.

Just a thought.  Getting back to the current structure, I agree that it looks a little odd with whitespace separating the columns, as opposed to standard HTML, which would have individual tags around every element (`<td>`).  Writing out `<pl-data>` or something for every entry would get very tedious, but I think it would be more standard.

Reply:
I agree that the whitespace isn't the best, and only came to implement it to avoid requiring something tedious like the `pl-data` tags. I hadn't considered the `show-rows` list attribute, however, and think that that's a great idea!

Reply:
So after looking at the code, I think that the whitespace is not necessary and can be ignored by the code parsing the row tags. It seems like using the `show-rows` attribute as described could make things more difficult for the user, as it requires translation from a bitstring to decimal.

Instead, if the `pl-row` element ignores whitespace, inputs could look like `<pl-row>00101</pl-row>`, with the option to add spaces if the developer finds that easier to read.

Reply:
To me, this is even more of an argument that there should be more explicit tags or separators here, if we are going to have the feature of explicit row specification at all.

Honestly, I think it makes more sense to ship this element _without_ this feature at all, or with the `show-rows` thing that Nathan suggested above. 

Then, after people start using it, if lots of people want this feature where you can explicitly define the rows you want to have and not have, it could be added as a feature.

Reply:
@stac21 Do you find this row selection feature useful in the use cases you have come across? I think it's easy enough to add / remove as-needed, so if people think it's a useful feature then it's worth shipping with the initial version of this element.

Reply:
That's the thing though--once you ship a feature like this to production, its _not_ easy to remove it, because you will break people's existing questions. 

That's why I lean toward don't add the feature until later, when we know people will actually use it.